plugins/terminal/ios.py

```diff
@@ -92,10 +116,13 @@ def on_become(self, passwd=None):
                 to_bytes(json.dumps(cmd), errors="surrogate_or_strict")
             )
             prompt = self._get_prompt()
-            if prompt is None or not prompt.endswith(b"#"):
+            if (
+                prompt is None
+                or not prompt.endswith(b"#")
+                or self.get_privilege_level() != 15
+            ):
                 raise AnsibleConnectionFailure(
-                    "failed to elevate privilege to enable mode still at prompt [%s]"
-                    % prompt
+                    "failed to elevate privilege to enable mode still at prompt"
                 )
```

Comment:
So originally, this error told you why Ansible believed that the elevation failed (i.e., "still at prompt [device%]" or whatever. Now there are a variety of reasons we might think we're not at the right privilege level, and I think we should tell the user why we failed rather than just failing with a message that no longer gives the user any information about why.

Looking closer, we _also_ have the overlapping exception issue here. I would probably also retarget the exceptions so that we are not nesting exceptions unreasonably deeply. GitHub won't let me put a suggestion here, but basically turn lines 118 through 132 to:

```
            prompt = self._get_prompt()
            privilege_level = self.get_privilege_level()
        except AnsibleConnectionFailure as e:
            prompt = self._get_prompt()
            raise AnsibleConnectionFailure(
                "failed to elevate privilege to enable mode, at prompt [%s] with error: %s"
                % (prompt, e.message)
            )

        if (
            prompt is None
            or not prompt.endswith(b"#")
            or privilege_level != 15
        ):
            raise AnsibleConnectionFailure(
                "failed to elevate privilege to enable mode, still at level [%d] and prompt [%s]"
                % (privilege_level, prompt)
            )
```